contracts/governance/proposal/ProposalMasterCharlieV1.sol

```diff
@@ -0,0 +1,426 @@
+    uint256 public quorumPercentage;
+
+    /// @notice The period (expressed in blocks) before voting goes live for a
+    /// given proposal.
+    uint256 public proposalDelay;
+
+    /// @notice The period (expressed in blocks) in which voters are able to
+    /// cast a vote for a given proposal.
+    uint256 public proposalVotingPeriod;
+
+    /// @notice Emitted when the proposal threshold percentage is updated.
+    event ProposalPercentageUpdated(
+        uint256 oldPercentage,
+        uint256 newPercentage
+    );
+
+    /// @notice Emitted when the quorum threshold percentage is updated.
+    event QuorumPercentageUpdated(
+        uint256 oldPercentage,
+        uint256 newPercentage
+    );
+
+    /// @notice Emitted when a new governance token is added.
+    event TokenAdded(address token);
+
+    /// @notice Emitted when a governance token is removed.
+    event TokenRemoved(address token);
+
+    /// @notice Emitted when an account's balance is excluded from being counted
+    /// towards a token's total supply.
+    event ExclusionAccountAdded(address token, address account);
+
+    /// @notice Emitted when an account's balance is no longer excluded from
+    /// being counted towards a token's total supply.
+    event ExclusionAccountRemoved(address token, address account);
+
+    /**
+     * @param tokens_ The initial list of governance tokens to support.
+     * @param timelock The timelock address associated with governance.
+     * @param governance_ The master governance contract address.
+     * @param initialProposalPercentage The percentage to use as threshold for
+     * checking proposal creation eligibility.
+     * @param initialQuorumPercentage The percentage to use for calculating the
+     * required quorum for passing a proposal.
+     */
+    constructor(
+        address[] memory tokens_,
+        address timelock,
+        address governance_,
+        uint256 initialProposalPercentage,
+        uint256 initialQuorumPercentage
+    ) {
+        for (uint256 i = 0; i < tokens_.length; i++) {
+            addToken(tokens_[i]);
+        }
+
+        governance = IGovernor(governance_);
+        updateProposalPercentage(initialProposalPercentage);
+        updateQuorumPercentage(initialQuorumPercentage);
+        transferOwnership(timelock);
+    }
+
+    /**
+     * @notice Update the percentage value used to calculate the voting power
+     * required to create a new proposal.
+     * @param percentage The new percentage to use.
+     */
+    function updateProposalPercentage(uint256 percentage) public onlyOwner {
+        require(
+            percentage <= 100,
+            "SimpleProposalMaster::setPercentage: Invalid percentage"
+        );
+
+        require(
+            percentage != proposalPercentage,
+            "SimpleProposalMaster::setPercentage: Value is already in use"
+        );
+
+        emit ProposalPercentageUpdated(proposalPercentage, percentage);
```

Comment:
maybe nitpicky, but set before emit event?

Reply:
Nitpicks aren't bad; they might lead to some interesting discoveries.

My original reasoning for this format was as follows:
 1. setting the new percentage before emitting an event means using a temporary variable to store the old percentage to pass to the event; and
 2. setting an extra variable means extra gas expenditure.

However, I checked anyway (using [this gist](https://gist.github.com/JackieXu/351d8dddac4c965adcc4876519737f72)), and it's actually 9 gas cheaper with the temp variable in between. Tested on `0.8.4+commit.c7e474f2` with 200 optimizations and without.

Good observation, I'll update the functions accordingly. I like the readability better with the event emission on the bottom.